components/suggestion/command_provider/command_provider.test.tsx

```diff
@@ -0,0 +1,193 @@
+                    name: 'current_user_id__existingId',
+                    display_name: 'Default',
+                    delete_at: 0,
+                    type: '0',
+                    total_msg_count: 0,
+                    team_id: 'team_id',
+                },
+            },
+            channelsInTeam: {
+                'team-id': ['current_channel_id'],
+            },
+        },
+        teams: {
+            currentTeamId: 'team-id',
+            teams: {
+                'team-id': {
+                    id: 'team_id',
+                    name: 'team-1',
+                    displayName: 'Team 1',
+                },
+            },
+            myMembers: {
+                'team-id': {roles: 'team_role'},
+            },
+        },
+        users: {
+            currentUserId: 'current_user_id',
+            profiles: {
+                current_user_id: {roles: 'system_role'},
+            },
+        },
+        preferences: {
+            myPreferences: {
+                'display_settings--name_format': {
+                    category: 'display_settings',
+                    name: 'name_format',
+                    user_id: 'current_user_id',
+                    value: 'username',
+                },
+            },
+        },
+        roles: {
+            roles: {
+                system_role: {
+                    permissions: [],
+                },
+                team_role: {
+                    permissions: [],
+                },
+                channel_role: {
+                    permissions: [],
+                },
+            },
+        },
+        general: {
+            license: {IsLicensed: 'false'},
+            serverVersion: '5.4.0',
+            config: {PostEditTimeLimit: -1},
+        },
+        posts: {
+            posts: {
+                aaa: {},
+            },
+        },
+    },
+    views: {
+        rhs: {
+            selectedPostId: 'aaa',
+        },
+    },
+};
+
+describe('CommandSuggestion', () => {
+    const baseProps = {
+        item: {
+            suggestion: '/invite',
+            hint: '@[username] ~[channel]',
+            description: 'Invite a user to a channel',
+            iconData: '',
+        },
+        isSelection: true,
+        term: '/',
+        matchedPretext: '',
+    };
+
+    test('should match snapshot', () => {
+        const wrapper = shallow(
+            <CommandSuggestion {...baseProps}/>,
+        );
+
+        expect(wrapper).toMatchSnapshot();
+        expect(wrapper.find('.slash-command__title').first().text()).toEqual('invite @[username] ~[channel]');
+        expect(wrapper.find('.slash-command__desc').first().text()).toEqual('Invite a user to a channel');
+    });
+});
+
+describe('CommandProvider', () => {
+    const makeStore = async () => {
+        const initialState = {
+            ...reduxTestState,
+        } as any;
+        const testStore = await mockStore(initialState);
```

Comment:
Does configureStore can use parameters, so a initialState of the proper type can be used.

Reply:
Sorry, my English sometimes break :sweat_smile: Here is an example of what I mean:
https://github.com/mattermost/mattermost-webapp/blob/3623ebadc29253ca9c12abbd41f095caa2a1c620/actions/views/rhs.test.ts#L39

Reply:
Now, in theory, you don't have to convert initial state to any for this to work.

Reply:
@larkox I changed it from `any` to `GlobalState`. Without any type assertion, I get this error:

```
components/suggestion/command_provider/command_provider.test.tsx(152,43): error TS2345: Argument of type '{ entities: { channels: { currentChannelId: string; myMembers: { current_channel_id: { channel_id: string; user_id: string; roles: string; mention_count: number; msg_count: number; }; }; channels: { ...; }; channelsInTeam: { ...; }; }; ... 5 more ...; posts: { ...; }; }; views: { ...; }; }'
is not assignable to parameter of type 'GlobalState | MockGetState<GlobalState> | undefined'.
```

Reply:
@larkox Initializing a redux store actually wasn't needed for the tests written, since the global store is now being used, which is already initialized with values. No tests in the file rely on the redux state.